src/apple/disk.rs

```diff
@@ -53,14 +62,330 @@ impl DiskExt for Disk {
 
     fn refresh(&mut self) -> bool {
         unsafe {
-            let mut stat: statfs = mem::zeroed();
-            let mount_point_cpath = to_cpath(&self.mount_point);
-            if statfs(mount_point_cpath.as_ptr() as *const i8, &mut stat) == 0 {
-                self.available_space = u64::from(stat.f_bsize).saturating_mul(stat.f_bavail);
-                true
+            let requested_properties = build_requested_properties(&[
+                ffi::kCFURLVolumeAvailableCapacityKey,
+                ffi::kCFURLVolumeAvailableCapacityForImportantUsageKey,
+            ]);
+            match get_disk_properties(&self.volume_url, &requested_properties) {
+                Some(disk_props) => {
+                    self.available_space = get_available_volume_space(&disk_props);
+                    true
+                }
+                None => false,
+            }
+        }
+    }
+}
+
+pub(super) fn get_disks() -> Vec<Disk> {
+    let raw_disks = unsafe {
+        let count = libc::getfsstat(ptr::null_mut(), 0, libc::MNT_NOWAIT);
+        if count < 1 {
+            return Vec::new();
+        }
+        let bufsize = count * std::mem::size_of::<libc::statfs>() as libc::c_int;
+        let mut disks = Vec::with_capacity(count as _);
+        let count = libc::getfsstat(disks.as_mut_ptr(), bufsize, libc::MNT_NOWAIT);
+
+        if count < 1 {
+            return Vec::new();
+        }
+
+        disks.set_len(count as usize);
```

Comment:
```suggestion
        disks.set_len(count as _);
```

Reply:
I just had to fix clippy lints related to unneeded casts because a type changed because of a platform change that I'd much rather prefer to have a generic cast.

Reply:
This one isn't platform specific, it comes out of the Rust standard library. It would be a breaking change if they made this not take a `usize` so I am not sure that I have the same level of worry with this one.

Reply:
Oh indeed. Sorry, I don't why I was sure it was another type. My bad. Resolved then.